connector/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaOffsetRangeCalculator.scala

```diff
@@ -80,21 +83,39 @@ private[kafka010] class KafkaOffsetRangeCalculator(val minPartitions: Option[Int
         } else {
           getPartCount(size, splitRangeTotalSize, splitRangeMinPartitions)
         }
-        var remaining = size
-        var startOffset = range.fromOffset
-        (0 until parts).map { part =>
-          // Fine to do integer division. Last partition will consume all the round off errors
-          val thisPartition = remaining / (parts - part)
-          remaining -= thisPartition
-          val endOffset = math.min(startOffset + thisPartition, range.untilOffset)
-          val offsetRange = KafkaOffsetRange(tp, startOffset, endOffset, None)
-          startOffset = endOffset
-          offsetRange
-        }
+        getDividedPartition(parts, range)
+      }.filter(_.size > 0)
+    } else {
+      val avgMessageSize = 1 // Todo estimate row size
```

Comment:
Help required: How can we estimate avgMessage size here? If not shall we keep maxRecordsPerPartition config?

Reply:
It is the main challenge on this approach - if you don't have an idea to calculate this in general, the overall proposal just does not work.

Reply:
I am using `maxRecordsPerPartition` to dynamically increase partitions. PTAL. Thanks